### PR TITLE
60-add-search-of-user-dirictory

### PR DESCRIPTION
**Pull request by BtsuanEyakem**

### CODE REVIEW

File: src/main/java/edu/miu/cs545/project/repository/StudentDirectoryRepo.java [hunk 1]
```diff
@@ -4,6 +4,27 @@
...
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Optional;
+
 @Repository
 public interface StudentDirectoryRepo extends ListCrudRepository<StudentDirectory, Long> {
+    Optional<List<StudentDirectory>> findByMajor(String major);
+    Optional<List<StudentDirectory>> findByAcademicYear(int date);
+
+    Optional<List<StudentDirectory>> findByUser_FirstNameOrUser_LastNameOrUser_UsernameOrUserEmailOrUser_PhoneOrUser_AddressOrUser_CityOrUser_StateOrUser_ZipOrUser_CountryOrUser_Department(String firstName,
```
Comment (nabil-miu) [inline src/main/java/edu/miu/cs545/project/repository/StudentDirectoryRepo.java]: @BtsuanEyakem can we use Critieria Builder to do the search ?
https://www.baeldung.com/spring-data-criteria-queries

File: src/main/java/edu/miu/cs545/project/mapper/impl/UserDtoMapper.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
+package edu.miu.cs545.project.mapper.impl;
+
+import edu.miu.cs545.project.dto.UserDTO;
+import edu.miu.cs545.project.model.entity.User;
+
+import java.time.LocalDate;
+
+public class UserDtoMapper{
```
Comment (nabil-miu) [inline src/main/java/edu/miu/cs545/project/mapper/impl/UserDtoMapper.java]: can we use mapstruct mapper? i mean having an interface and implementing it ?

File: src/main/java/edu/miu/cs545/project/service/impl/SurveyServiceImpl.java [hunk 3]
```diff
@@ -3,12 +3,26 @@
...
 import org.springframework.data.repository.ListCrudRepository;
 import org.springframework.stereotype.Service;
 
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+
 @Service
 public class SurveyServiceImpl extends CrudServiceImpl<Survey,Long> implements SurveyService {
+    @Autowired
```
Comment (darkhanzhursin) [inline src/main/java/edu/miu/cs545/project/service/impl/SurveyServiceImpl.java]: remove annotation

File: src/main/java/edu/miu/cs545/project/service/impl/StudentDirectoryServiceImpl.java [hunk 4]
```diff
@@ -3,11 +3,42 @@
...
 import org.springframework.stereotype.Service;
 
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Optional;
+
 @Service
 public class StudentDirectoryServiceImpl extends CrudServiceImpl<StudentDirectory, Long> implements StudentDirectoryService {
+
+    @Autowired
```
Comment (darkhanzhursin) [inline src/main/java/edu/miu/cs545/project/service/impl/StudentDirectoryServiceImpl.java]: remove annotation

File: src/main/java/edu/miu/cs545/project/controller/SurveyController.java [hunk 5]
```diff
@@ -3,15 +3,29 @@
...
 import org.springframework.web.bind.annotation.RestController;
 
+import java.util.List;
+
 @RestController
 @RequestMapping("/api/v1/surveys")
 @Tag(name = "Surveys", description = "Surveys API")
 public class SurveyController extends CrudController<Survey, Long> {
 
+    @Autowired
```
Comment (darkhanzhursin) [inline src/main/java/edu/miu/cs545/project/controller/SurveyController.java]: remove annotation

**Pull request closed without merging**